test_runner/regress/test_tenant_size.py

```diff
@@ -9,28 +9,233 @@ def test_empty_tenant_size(neon_simple_env: NeonEnv):
     env = neon_simple_env
     (tenant_id, _) = env.neon_cli.create_tenant()
     http_client = env.pageserver.http_client()
-    size = http_client.tenant_size(tenant_id)
+    initial_size = http_client.tenant_size(tenant_id)
 
-    # we should never have zero, because there should be the initdb however
-    # this is questionable if we should have anything in this case, as the
-    # gc_cutoff is negative
-    assert (
-        size == 0
-    ), "initial implementation returns zero tenant_size before last_record_lsn is past gc_horizon"
+    # we should never have zero, because there should be the initdb "changes"
+    assert initial_size > 0, "initial implementation returns ~initdb tenant_size"
 
-    with env.postgres.create_start("main", tenant_id=tenant_id) as pg:
+    main_branch_name = "main"
+
+    with env.postgres.create_start(main_branch_name, tenant_id=tenant_id) as pg:
         with pg.cursor() as cur:
             cur.execute("SELECT 1")
             row = cur.fetchone()
             assert row is not None
             assert row[0] == 1
         size = http_client.tenant_size(tenant_id)
-        assert size == 0, "starting idle compute should not change the tenant size"
+        assert size == initial_size, "starting idle compute should not change the tenant size"
 
     # the size should be the same, until we increase the size over the
     # gc_horizon
     size = http_client.tenant_size(tenant_id)
-    assert size == 0, "tenant_size should not be affected by shutdown of compute"
+    assert size == initial_size, "tenant_size should not be affected by shutdown of compute"
```

Comment:
I am wondering if we should pick out the model inputs for this case, check the updates which are assumed to be:

- lsn: 0, command: BranchFrom(None), timeline: timeline_id
- lsn: initdb_lsn, command: EndOfBranch, timeline: timeline_id

Reply:
I think it's enough to test user-visible behavior in these tests.
We may write similar unit-test to cover model-inputs later.

Reply:
Well, I ended up writing it since at least the comment was already in the fixture/support code :) (It was the initial idea to start asserting those, but then I decided against it, and didn't revisit this when the EndOfBranch was added.)

Please see: https://github.com/neondatabase/neon/pull/3377/commits/7732e4ee7402782c09b32a11186841de5e24a010

Reply:
Small mypy issue, fixing that...

Reply:
cool!
 I expected that it will require more work)